coderd/authorize.go

```diff
@@ -10,9 +10,27 @@ import (
 	"github.com/coder/coder/coderd/rbac"
 )
 
-func AuthorizeFilter[O rbac.Objecter](api *API, r *http.Request, action rbac.Action, objects []O) []O {
+func AuthorizeFilter[O rbac.Objecter](api *API, r *http.Request, action rbac.Action, objects []O) ([]O, error) {
 	roles := httpmw.AuthorizationUserRoles(r)
-	return rbac.Filter(r.Context(), api.Authorizer, roles.ID.String(), roles.Roles, action, objects)
+
+	if len(objects) == 0 {
+		return objects, nil
+	}
+	objecType := objects[0].RBACObject().Type
+	objects, err := rbac.Filter(r.Context(), api.Authorizer, roles.ID.String(), roles.Roles, action, objecType, objects)
```

Comment:
Seems like all of this logic could be in the `rbac` package itself, then we wouldn't need the `recordingAuthorizer` code. Then functions could just call `rbac.AuthorizeFilter` directly.

Reply:
You are right about the object type stuff :+1: 

As for `recordingAuthorizer`. That is needed for my unit test that checks all the routes. `recordingAuthorizer` is never used outside unit tests.